flink-runtime/src/main/java/org/apache/flink/runtime/state/KeyGroupsStateHandle.java

```diff
@@ -91,10 +98,10 @@ public KeyGroupsStateHandle getKeyGroupIntersection(KeyGroupRange keyGroupRange)
 
 	/**
 	 *
-	 * @return the internal key-group range to offsets metadata
+	 * @return the start key group in the key-group range of this handle
 	 */
-	public KeyGroupRangeOffsets getGroupRangeOffsets() {
-		return groupRangeOffsets;
+	public int getStartKeyGroup() {
```

Comment:
I suggest to remove this pass-through method, as it at a very low abstraction level. This information can be obtained already from the result of `getKeyGroupRange()`. Now, that we expose the `KeyGroupRange`, also `getNumberOfKeyGroups()` could be removed, as we can ask the `KeyGroupRange` directly.